dev/tools/bin/generate_gradle_lockfiles.dart

```diff
@@ -41,8 +41,15 @@ void main(List<String> arguments) {
           'Run the script using the config file at ./configs/lockfile_exclusion.yaml to skip the specified subdirectories.',
       defaultsTo: true,
     )
-    ..addFlag('ignore-locking', help: 'Generate ignore file to disable gradle dependency locking. A reason must be given.')
-    ..addOption('ignore-reason', help: 'Reason to disable gradle dependency locking. A reason must be given if --ignore-locking is used.');
+    ..addFlag(
+      'ignore-locking',
+      help: 'Generate ignore file to disable gradle dependency locking. A reason must be given.',
+    )
+    ..addOption(
```

Comment:
You can simplify this code by only using ignore-reason. There is no reason to use the ignore-reason flag without ignore-locking and there is no reason to use ignore-locking without a reason.